Nexcloud stop to work while opening PDF Files

Hi,
We have just install a Nextcloud Server to change our owncloud solution but when using the last nextcloud playstore version on a samsung android tablet (android 7) the application stopped while opening or scrolling over a PDF file. Same issue with a samsung android phone. Can't check the Log as you removed the expert mode. So I tried to take an older APK : nextcloud-30020199.apk and it's working fine with it. Could you please tell us if it's a known problem which will be fixed because we will have a lot of android with nextcloud using google MDM to deploy and we can't do it with an old APK... Attached the error when application stop (in french sorry)

Best Regards
![Screenshot_20190509](https://user-images.githubusercontent.com/20945460/57460135-d82b4d00-7274-11e9-90fd-2313a54e0e8f.jpg)

This is the relevant part of the log:
```
05-10 16:01:03.635 27435 27435 I art     : Rejecting re-init on previously-failed class java.lang.Class<androidx.core.view.ViewCompat$OnUnhandledKeyEventListenerWrapper>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/View$OnUnhandledKeyEventListener;
05-10 16:01:03.635 27435 27435 I art     :   at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2203)
05-10 16:01:03.635 27435 27435 I art     :   at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:637)
05-10 16:01:03.635 27435 27435 I art     :   at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:518)
05-10 16:01:03.635 27435 27435 I art     :   at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:466)
05-10 16:01:03.635 27435 27435 I art     :   at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:140)
05-10 16:01:03.635 27435 27435 I art     :   at void com.owncloud.android.ui.activity.FileDisplayActivity.onCreate(android.os.Bundle) (FileDisplayActivity.java:239)
05-10 16:01:03.635 27435 27435 I art     :   at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:6948)
05-10 16:01:03.635 27435 27435 I art     :   at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1126)
05-10 16:01:03.635 27435 27435 I art     :   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2924)
05-10 16:01:03.635 27435 27435 I art     :   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:3042)
05-10 16:01:03.635 27435 27435 I art     :   at void android.app.ActivityThread.-wrap14(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
05-10 16:01:03.635 27435 27435 I art     :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1639)
05-10 16:01:03.635 27435 27435 I art     :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
05-10 16:01:03.635 27435 27435 I art     :   at void android.os.Looper.loop() (Looper.java:154)
05-10 16:01:03.635 27435 27435 I art     :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6780)
05-10 16:01:03.635 27435 27435 I art     :   at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
05-10 16:01:03.635 27435 27435 I art     :   at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:1500)
05-10 16:01:03.635 27435 27435 I art     :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1390)
05-10 16:01:03.635 27435 27435 I art     : Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/com.nextcloud.client-1/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-1/lib/arm64, /data/app/com.nextcloud.client-1/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]
05-10 16:01:03.635 27435 27435 I art     :   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:56)
05-10 16:01:03.635 27435 27435 I art     :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:380)
05-10 16:01:03.635 27435 27435 I art     :   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
05-10 16:01:03.635 27435 27435 I art     :   at void androidx.core.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, androidx.core.view.OnApplyWindowInsetsListener) (ViewCompat.java:2203)
05-10 16:01:03.635 27435 27435 I art     :   at android.view.ViewGroup androidx.appcompat.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:637)
05-10 16:01:03.635 27435 27435 I art     :   at void androidx.appcompat.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:518)
05-10 16:01:03.635 27435 27435 I art     :   at void androidx.appcompat.app.AppCompatDelegateImpl.setContentView(int) (AppCompatDelegateImpl.java:466)
05-10 16:01:03.635 27435 27435 I art     :   at void androidx.appcompat.app.AppCompatActivity.setContentView(int) (AppCompatActivity.java:140)
05-10 16:01:03.635 27435 27435 I art     :   at void com.owncloud.android.ui.activity.FileDisplayActivity.onCreate(android.os.Bundle) (FileDisplayActivity.java:239)
05-10 16:01:03.635 27435 27435 I art     :   at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:6948)
05-10 16:01:03.635 27435 27435 I art     :   at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1126)
05-10 16:01:03.636 27435 27435 I art     :   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2924)
05-10 16:01:03.636 27435 27435 I art     :   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:3042)
05-10 16:01:03.636 27435 27435 I art     :   at void android.app.ActivityThread.-wrap14(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
05-10 16:01:03.636 27435 27435 I art     :   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1639)
05-10 16:01:03.636 27435 27435 I art     :   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
05-10 16:01:03.636 27435 27435 I art     :   at void android.os.Looper.loop() (Looper.java:154)
05-10 16:01:03.636 27435 27435 I art     :   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6780)
05-10 16:01:03.636 27435 27435 I art     :   at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
05-10 16:01:03.636 27435 27435 I art     :   at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:1500)
05-10 16:01:03.636 27435 27435 I art     :   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1390)
05-10 16:01:03.636 27435 27435 I art     :
```

--- a/src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java
+++ b/src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java
@@ -841,10 +841,11 @@ public final class ThumbnailsCacheManager {
         protected void onPostExecute(Drawable drawable) {
             if (drawable != null) {
                 AvatarGenerationListener listener = mAvatarGenerationListener.get();
-                AvatarGenerationTask avatarWorkerTask = getAvatarWorkerTask(mCallContext);
-
-                if (this == avatarWorkerTask && listener.shouldCallGeneratedCallback(mUserId, mCallContext)) {
-                    listener.avatarGenerated(drawable, mCallContext);
+                if (listener != null) {
+                    AvatarGenerationTask avatarWorkerTask = getAvatarWorkerTask(mCallContext);
+                    if (this == avatarWorkerTask && listener.shouldCallGeneratedCallback(mUserId, mCallContext)) {
+                        listener.avatarGenerated(drawable, mCallContext);
+                    }
                 }
             }
         }

